oak-run-commons/src/main/java/org/apache/jackrabbit/oak/indexversion/IndexVersionOperation.java

```diff
@@ -75,7 +78,7 @@ public String toString() {
      */
     public static List<IndexVersionOperation> generateIndexVersionOperationList(NodeState rootNode, String parentPath,
                                                                                 List<IndexName> indexNameObjectList, long purgeThresholdMillis) {
-        return generateIndexVersionOperationList(rootNode, parentPath, indexNameObjectList, purgeThresholdMillis, true);
+        return generateIndexVersionOperationList(rootNode, parentPath, indexNameObjectList, purgeThresholdMillis, true, false);
```

Comment:
Shouldn't there be two different implementations for lucene and elastic (different classes).

I think we already have an issue to have similar hidden mounts for elastic index, to make elastic compatible with blue-green deployment

Reply:
Made the changes to make it into 2 separate implementations so that the logic is a bit more clean